### DISCUSSION TITLE
Is it possible to have an affix file associated with the personal dictionary?

### DISCUSSION THREAD

**Discussion by BlueCutOfficial** [retextjs/retext#90] [Q&A] [answered]
Hello, 

I would like to know it it's possible to have an affix document associated with the personal dictionary?

**I tend to think not because...**

When I read `retext-spell` and `nspell` documentation...

> `options.dictionary` A dictionary, `options.personal` Personal dictionary
> https://github.com/retextjs/retext-spell#optionsdictionary

> dictionaries (Array.<Dictionary>) — List of dictionary objects. The first must have an aff key, other aff keys are ignored
> => https://github.com/wooorm/nspell#nspelldictionary

> Personal dictionaries are not intertwined with affix document.
> https://github.com/wooorm/nspell#personal-dictionary-documents

... I kind of understand there can be only one affix document and that the personal dictionary can only contain a list of words. But I am not sure if my understanding is correct, because the example I see in the third link is not quite clear to me:
```
bar/baz
```
> `bar` is added as well, but modeled after the existing word `baz`

I don't understand what this sentence means and how you can define stuff like `bar/baz` in a personal dictionary.

**I am asking because...**

I use `retext-spell` to check markdown files in my native language, but it is technical content and there are a lot of English words that we commonly use in tech, and they can be singular or plural depending on the context. For instance, I would like to authorize "decorator"/"decorators" or "helper"/"helpers". It wouldn't be relevant to translate these tech words, but they don't exist in the French dictionary I use. So far I couldn't figure out how to use syntax like `helper/S` in the personal dictionary.

Should I define all the versions of a word I need or is there actually a way to configure things like singular/plural in a personal dictionary? 🙏

**Comment by wooorm** [comment, marked as answer]
Hey!

No, not possible. `.dic` and `.aff` files are tightly integrated with each other. If you want a real `.aff` file, you perhaps need a fork of your french dictionary.

That being said, personal dictionaries work by *modelling* words: https://github.com/wooorm/nspell#personal-dictionary-documents.
So you *model* something after a whole entry.

What that means is that, assuming `Simon` is in the dictionary you are using and has a bunch of rules attached to it, you can add `Titus/Simon` to model `Titus` (not in dic) with the same rules that `Simon` (in dic) gets.

So that practically means you need to find an word in your French dictionary that has an optional `s` suffix, and use that as the model!

**Reply by BlueCutOfficial** [reply]
Ok, super clear, thanks a lot for the explanation 👍
